pkg/restapi/query_runner.go

```diff
@@ -148,15 +149,19 @@ func (h *queryHandler) runQuery(w http.ResponseWriter, r *http.Request) {
 	run.Output = output
 	run.ExecutionTime = timeElapsed
 
-	err = h.runRepo.Create(run)
-	if err != nil {
-		zlog.Error().Err(err).Interface("model", run).Msg("a run cannot be saved")
-		writeError(w, "internal error", http.StatusInternalServerError)
+	zlog.Info().Str("id", run.ID).Dur("elapsed", timeElapsed).Msg("processed a new run")
 
-		return
-	}
+	if req.SaveRunInfo {
```

Comment:
It might be a good idea to remove run ID from the response if `SaveRunInfo` is `false`. We will let the client know that the execution results have not been saved.